im3components/wrf_tell/wrf_tell_counties.py

```diff
@@ -0,0 +1,204 @@
+    df.to_csv(name, index=False)
+
+
+def process_time_slice(
+        df: pd.DataFrame,
+        wrf_variables: List[str],
+        precisions: List[int],
+        mapping: pd.DataFrame,
+        output_path: str,
+):
+    write_output_file(
+        compute_county_weighted_mean(
+            mapping.merge(df, how='left', left_on='cell_index', right_index=True),
+            wrf_variables,
+            precisions,
+        ),
+        df.time.iloc[0],
+        output_path,
+    )
+
+
+def wrf_to_tell_counties(
+        wrf_file: str,
+        wrf_variables: List[str],
+        precisions: List[int],
+        county_shapefile: str = './Geolocation/tl_2020_us_county/tl_2020_us_county.shp',
+        weight_and_mapping_file: str = './grid_cell_to_county_weight.parquet',
+        output_path: str = './County_Output_Files',
+        n_jobs: int = 1,
+) -> None:
+    """
+    Aggregate WRF output data to county level using area weighted average.
+
+    :param str wrf_file: path to the WRF output file to aggregate
+    :param list(str) wrf_variables: list of variables to aggregate
+    :param list(int) precisions: list of precisions corresponding to the variables to aggregate
+    :param str county_shapefile: path to a shapefile (.shp) with county geometries
+    :param str weight_and_mapping_file: path to read or write a weights file which maps WRF grid cell to county weight
+    :param str output_path: path to which output should be written
+    :param n_jobs: number of time slices to process in parallel
+    """
+
+    begin_time = datetime.datetime.now()
+
+    if not isfile(wrf_file):
+        print('No file to process, exiting...')
+        exit()
+
+    wrf = salem.open_wrf_dataset(wrf_file)
+
+    # if there's not already a mapping file, create one
+    if not isfile(weight_and_mapping_file):
+
+        # using the first file and time:
+        # * get the crs
+        # * create the mapping of cell index to county and weight
+        wrf_crs = wrf.pyproj_srs
+        wrf_df = wrf[wrf_variables].isel(time=0).to_dataframe().reset_index(drop=True)
+        wrf_df = gpd.GeoDataFrame(wrf_df, geometry=wrf.salem.grid.to_geometry().geometry).set_crs(wrf_crs)
+        wrf_df['cell_index'] = wrf_df.index.values
+
+        # load the counties and reproject to WRF projection
+        counties = gpd.read_file(county_shapefile)[["GEOID", "geometry"]].rename(columns={
+            'GEOID': 'FIPS',
+        }).to_crs(wrf_crs)
+
+        # find the intersection between counties and wrf cells
+        intersection = gpd.overlay(counties, wrf_df, how='intersection')
+        # weight by the intersection area
+        intersection['area'] = intersection.area
+        intersection['weight'] = (
+            intersection['area'] / intersection[['FIPS', 'area']].groupby('FIPS').area.transform('sum')
+        )
+
+        # reuse this mapping for the remaining files and slices
+        mapping = intersection[['cell_index', 'FIPS', 'weight']]
```

Comment:
check that `.copy()` should not be used in this case

Reply:
since i'm not further manipulating this frame, i think the view is okay